tests/bcr/go.mod

```diff
@@ -1 +1,5 @@
-// Only here to stop go mod from descending into this directory.
\ No newline at end of file
+// Only here to stop go mod from descending into this directory.
+// We will run `go mod tidy` to make sure all checksums are properly evaluated.
```

Comment:
How are we doing that? ;-)

Reply:
Removed this. We can adjust and make a better test for the `go.mod` logic in a separate PR. 

This BCR test will still use the `go.sum` file in the root of the repo to validate, since we load gazelle as a `bazel_dep`.